src/autofill-utils.js

```diff
@@ -69,45 +70,68 @@ const setValueForInput = (el, val) => {
     originalSet.call(el, val)
     events.forEach((ev) => el.dispatchEvent(ev))
     el.blur()
+
+    return true
+}
+
+/**
+ * Fires events on a select element to simulate user interaction
+ * @param {HTMLSelectElement} el
+ */
+const fireEventsOnSelect = (el) => {
+    const events = [
+        new Event('mousedown', {bubbles: true}),
+        new Event('focus', {bubbles: true}),
+        new Event('change', {bubbles: true}),
+        new Event('mouseup', {bubbles: true}),
+        new Event('click', {bubbles: true})
+    ]
+    // Events fire on the select el, not option
+    events.forEach((ev) => el.dispatchEvent(ev))
+    events.forEach((ev) => el.dispatchEvent(ev))
+    el.blur()
 }
 
 /**
  * Selects an option of a select element
  * We assume Select is only used for dates, i.e. in the credit card
  * @param {HTMLSelectElement} el
  * @param {string | number} val
+ * @return {boolean}
  */
 const setValueForSelect = (el, val) => {
+    // Loop first through all values because they tend to be more precise
```

Comment:
We're now doing this loop twice to make sure we first evaluate all `value`s. The loop is cheap and the lists tend to be rather short.